projects/core/src/site-context/facade/site-context-persistence.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable, OnDestroy } from '@angular/core';
```

Comment:
I don't like the single context persistence service, rather have a factory/service per context (lang/curr). 

What about moving the sync logic to the `langService.initialize()` and `currService.initialize()`? We already have `initializeContext` factory that calls the `initialize()` on each context service, including the `BaseSiteService`. 

We can then add the sync feature to the existing initialise and drop the overhead of yet another factory.

Reply:
Moved to two "persistence" services